admin_guide/http_proxies.adoc

```diff
@@ -0,0 +1,167 @@
+----
+
+[[proxying-docker-pull]]
+
+== Proxying Docker Pull
+
+OpenShift node hosts need to perform push and pull operations to Docker
+registries. If you have a registry that does not need a proxy for nodes to
+access, include the `*NO_PROXY*` parameter with the registry's hostname, the
+registry service's IP address, and service name. This blacklists that registry's
+address, leaving the external http proxy as the only option.
+
+. Edit the *_/etc/sysconfig/docker_* file and add the variables in shell format:
++
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com,172.30.123.45,docker-registry.default.svc.cluster.local
+----
+====
+
+. Restart the Docker service:
++
+----
+# systemctl restart docker
+----
+
+////
+. If you had already imported ImageStreams before creating the proxy
+configuration, delete, then re-run the process:
++
+----
+# oc delete imagestreams -n openshift --all
+# oc create -f image-streams.json -n openshift
+----
+////
+
+[[s2i-builds]]
+
+== S2I Builds
+
+S2I builds fetch dependencies from various locations. You can
+link:../dev_guide/builds.html#environment-files[use a *_.sti/environment_* file]
+to specify simple shell variables and OpenShift will react accordingly when
+seeing build images.
+
+The following are the supported proxy environment variables with example values:
+
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+[[configuring-default-templates-for-proxies]]
+
+== Configuring Default Templates for Proxies
+
+The example templates available in OpenShift by default do not include settings
+for HTTP proxies. For existing applications based on these templates, modify the
+`*source*` section of the buildConfig and add proxy settings:
+
+====
+"source": {
+  "type": "Git",
+  "git": {
+    "uri": "https://github.com/openshift/ruby-hello-world.git",
+    "httpProxy"  : "http://proxy.example.com",
+    "httpsProxy" : "https://proxy.example.com"
+  }
+}
+====
+
+This is similar to the process for link:#git-repository-access[using proxies for
+Git cloning].
+
+[[setting-environment-variables-in-pods]]
+
+== Setting Environment Variables in Pods
+
+Set the following environment variables in a DeploymentConfig to pass database
+connection information. The same can be done for configuring a Pod's proxy at
+runtime:
+
+====
+----
+{
+  "apiVersion": "v1beta1",
+  "kind": "DeploymentConfig",
+  "metadata": {
+    "name": "frontend"
+  },
+  "template": {
+    "controllerTemplate": {
+      "podTemplate": {
+        "desiredState": {
+          "manifest": {
+            "containers": [
+              {
+                "env": [
+                  {
+                    "name": "HTTP_PROXY",
+                    "value": "http://USER:PASSWORD@IPADDR:PORT"
+                  },
+----
+====
```

Comment:
@adellape I'm concerned that the less savvy reader might think this is the entirety of the data structure, copy and paste it, and run into problems.  To avoid this, i suggest adding a title to the block w/ the word "fragment" in it.  (This is what we do in the GNU project, btw.)

To really drive the point home, we could indent the thing by four spaces (two for `template`, two for `spec`) and add `[...]` above and below.  That's really noisy, however, and probably only appropriate in contexts where the example is either alone on the page, or in the midst of complete examples that might mislead the reader into thinking this one is also complete.  I don't know if this is such a context; i'm just spewing ideas.  :smile:

This whole "how and when to portray excerpts" topic is something we might chew on at the next writer's meeting...